documents/design/principles.md

```diff
@@ -58,6 +58,24 @@ ICU4C/ICU4J exposes certain pieces of data through user-facing APIs such as Date
 
 Runtime customizability of locale data can sometimes come at a performance or memory cost.
 
+## Locale data from multiple sources works seamlessly
+
+*What:* If data is available for a particular constructor and locale from multiple data providers derived from the same CLDR version, then i18n correctness should not change based on which data provider is used to load the data.
```

Comment:
thought: should we explicitly mention that this is talking about _successful_ runs? We are okay with things erroring on checksum mismatch. There's an example on this below but it's not quite clear.